experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -2,36 +2,44 @@
 # 1: Construct auxiliary base space
 ################################################################
 
+
 function _auxiliary_base_space(auxiliary_base_variable_names::Vector{String}, auxiliary_base_grading::Matrix{Int64}, d::Int)
+
+  # We now try to guess one toric base space with the desired grading and maximal dimension
+  charges = matrix(ZZ, auxiliary_base_grading)
+  variety = normal_toric_variety_from_glsm(charges)
+  G1 = free_abelian_group(ncols(charges))
+  G2 = free_abelian_group(nrows(charges))
+  set_attribute!(variety, :map_from_torusinvariant_weil_divisor_group_to_class_group, hom(G1, G2, transpose(charges)))
+  set_attribute!(variety, :class_group, G2)
+  set_attribute!(variety, :torusinvariant_weil_divisor_group, G1)
   
-  # Find candidate base spaces
-  candidates = normal_toric_varieties_from_glsm(matrix(ZZ, auxiliary_base_grading))
-  @req length(candidates) > 0 "Could not find a full star triangulation"
-  f = candidates[1]
-  @req dim(f) >= d "Cannot construct an auxiliary base space of the desired dimension"
+  # Check if dimensional requirement is met
+  @req dim(variety) >= d "Cannot construct an auxiliary base space of the desired dimension"
   
   # Construct one base space of desired dimension
-  fan_rays = matrix(ZZ, rays(f))
-  fan_rays = [vec([Int(a) for a in fan_rays[k,:]]) for k in 1:nrows(fan_rays)]
-  fan_max_cone_matrix = matrix(ZZ, cones(f))
-  new_max_cones = Vector{Int}[]
-  for k in 1:nrows(fan_max_cone_matrix)
-    indices = findall(x -> x == 1, vec(fan_max_cone_matrix[k,:]))
-    if length(indices) == d
-      push!(new_max_cones, indices)
+  auxiliary_base_space = variety
+  if dim(auxiliary_base_space) != d
+    integral_rays = matrix(ZZ, rays(variety))
+    integral_rays = [vec([Int(a) for a in integral_rays[k,:]]) for k in 1:nrows(integral_rays)]
```

Comment:
I keep seeing this popping up. Casting to `Int` can result in a loss of precision. Why is this necessary?

Reply:
It has been a while. My recollection is that one of the functions that I am using required a matrix with `Int` entries. Maybe `convex_hull`. But then, your suggestion below (which looks great) avoids that altogether. So there is then probably no need for this cast anymore. Let me try out your suggestion.

Reply:
Perfect. Works!

Reply:
This is not resolved, since there was no suggestion here. Maybe something like:
```suggestion
```

Reply:
Sorry. My bad. I simply forgot to delete this line (as I see it, it is no longer needed). I just pushed the corresponding change.